### PR TITLE
Content: Creating light secondary weapon pods.

### PR DESCRIPTION
**Pull request by Zitchas**
**Content (Artwork / Outfit)**

## Summary
Adds:
- Meteor Missile Pod
- Sidewinder Missile Pod
- Javelin Mini Pod
- Heavy Rocket Pod
- Torpedo Pod
- Typhoon Pod

## Details
The current implementation of missiles causes some problems for combat balance in the game. The predominant one being that launchers are all balanced such to be a useful capacity and effectiveness when installed on warships, but are also installed on fighters and interceptors.

For example, the wasp (meteor) variant carries two launchers with 60 meteors, capable of doing some combination of 15'600 shield and 6'900 hull damage. That's enough to take a full strength falcon and reduce it to roughly half hull, for comparison. This seems excessive, to say the least. Even more ridiculous is the stock Flivver, whose description talks about lacking the space for heavy weaponry and only redeeming feature being speed - and yet it stocks the firepower to drive off one of the best civilian heavy warships. Of course, AM does come into play, and jamming/IR changes might apply too now. But nonetheless is utterly unbelievable that a stock civilian racing ship or light interceptor should be capable of fielding such massive destructive firepower.

Perhaps the strangest, though, is the case of the clipper - a light freighter that sometimes comes with 2 heavy rocket launchers and 40 heavy rockets, which totals up to some combination of up to 34k shields and 29k hull. Thus this humble freighter sometimes carries sufficient firepower to utterly annihilate *Carriers* (ignoring AM).

On the other hand, fighters commonly use missiles, and are often the delivery mechanism for heavy ordinances. Thus the dilemma that we need launchers suitable for use by fighters, drones, and light ships; but that don't fill them up with enough munitions to put them into full-on warship territory.

Thus the concept of weapon pods. Each of these pods is designed around the idea of being used as either:
a) main weapons on fighters, drones, and interceptors.
b) backup / secondary weapons for non-combat ships intended to provide covering fire while they escape.

It should also be worth noting that now that we have reloading mechanics for fighters/drones to reload from their carrier, having smaller capacities on the ships themselves is both more feasible and makes more sense.

These pods have been balanced in relation to current missile stats, but I expect them to shrink further to something close to their stats in the big combat Revamp #5772 if/when missiles are adjusted to be individually slower but deadlier.

Most of the graphics courtesy of Becca.
Torpedo and Typhoon pod courtesy of Ravenshining.

Copyright changes also include two re-orderings to be in alphabetical order.

## Testing
These outfits should be available wherever their standard launchers are available.

## PR Checklist
 - [X] I updated the copyright attributions, or decline to claim copyright of any assets produced or modified
 - [X] I uploaded the necessary image, blend, and texture assets here: [zip file on google drive](https://drive.google.com/file/d/1kRl_Rbu724sJgTAZ5xflXPSratfeiPJF/view?usp=sharing)
 - [X] I created a PR to the [endless-sky-high-dpi repo](https://github.com/endless-sky/endless-sky-high-dpi) with the `@2x` versions of these art assets: [https://github.com/endless-sky/endless-sky-high-dpi/pull/169](https://github.com/endless-sky/endless-sky-high-dpi/pull/169)

### CODE REVIEW

**Comment by Zitchas** [comment]
It is anticipated that after the outfits themselves are merged, the next step would be to implement their use as stock equipment to replace the full-size launchers on drones, fighters, and interceptors. Of particular importance is those ships that frequently turn up as pirates in the early-game core, dirt belt, and paradise world areas. (aka, the new-pilot killers).

From a lore perspective, if you are a pirate lord sending poorly trained teenagers into the core to raid for you, are you really going to pay for a full sized launcher that carries so much ammo your crew probably won't even spend half of before they die? Or are you going to send them with just enough ammo to drive off one or two fighters and disable a light freighter? Anything more than that is just throwing money down the drain.

File: data/human/weapons.txt [hunk 1]
```diff
@@ -693,6 +693,45 @@ outfit "Meteor Missile Launcher"
...
+		"drag" .1
+		"turn" 1.8
+		"homing" 1
+		"infrared tracking" .65
+		"shield damage" 260
+		"hull damage" 160
+		"hit force" 220
+		"missile strength" 6
+	description "The Meteor Missile is the cheapest and simplest homing weapon available in human space. Meteors are quickly destroyed by anti-missile systems, and their infrared tracking systems have trouble homing in on targets with low heat levels. Nevertheless, their large warheads can spell doom for poorly-defended targets when the missiles are fired in great numbers."
+	description "	Pods are designed to enable fighters and other light craft to carry small quantities of munitions into battle. While they are quite efficient at launching their normal capacity, the reduced size results in a less efficient loader that doesn't accommodate boxes as well as a normal launcher."
```
Comment (Pointedstick) [inline data/human/weapons.txt:733]: Maybe something a bit simpler:

```suggestion
	description "	Pods enable fighters and other light craft to carry small quantities of missiles into battle while retaining enough space for other weapons. Large ships are better off with full-size launchers."
```

**Comment by Pointedstick** [comment]
Just what I had in mind. Looks great!

**Comment by Pointedstick** [comment]
Gotta change the other ones too :)

**Comment by Pointedstick** [comment]
No Torpedo Pod? Did you think it overlapped in intended use too much with the Rocket Pod?

**Comment by Pointedstick** [comment]
This will fix https://github.com/endless-sky/endless-sky/issues/3372 BTW.

**Comment by Zitchas** [comment]
I don't have time right now to edit the rest of the text, but I will within the next 24h.

I don't have a torpedo tube in here mainly because @beccabunny didn't want to make one; although some people felt that it was too much of a "Big Missile for Capital-on-Capital warfare" thing that shouldn't really be on fighters. 

There's arguments either way, but felt that things might go better if we do the pods for the weapons that everyone pretty much agrees on first, then do others separately.

**Comment by Terin** [comment]
These are definitely great for giving players more flexibility in terms of outfitter fighters or smaller crafts. However, in the case of heavy rockets, I want us to make sure they won't become more prevalent on fast ships when starting out, as they can be an extremely frustrating weapon that one-shots the player. I'd almost rather heavy rockets be exclusively an anti-fighter swarm weapon deployed by light/medium warships, and torpedoes as anti-capital weapons. A Fury being able to deal 3400 damage in a single salvo, and over 10k overall feels excessive.

**Comment by Zitchas** [comment]
@Terin  Definitely. I'm not changing any loadouts in this PR, but my intent with future PRs is that most *existing* ,non-plot, variant drone, fighter, or interceptor that has a full size launcher will be swapped for the pod version. I do not intend to add any launchers to stock or variants that do not already have them. Possible exception of the javelin mini pod. So basically, the "total damage output" for all the drones, fighters, and interceptors should go down.

*exception being mission-specific ships, as in custom variants specific for specific missions, which I don't plan on touching at all.

**Comment by Pointedstick** [comment]
I don't think we would change any fleets to have quad rocket Furies by default. However the effectiveness of such a ship shows exactly why making it possible is necessary. This is exactly why fighter-sized craft carry heavy munitions in the first place: to be able to deliver a devastating pulse of firepower at a distance. Are you scared of this? better beef up your anti-missile defenses and equip some Sidewinder launchers.

Another thing that came up recently was the idea of reducing the trigger radius of the Heavy Rocket. I think reducing its speed would help too (it is _heavy_, after all), with a corresponding increase to anti-missile defense.

**Comment by Zitchas** [comment]
OK, the descriptions have been updated with the changes you suggested. Thanks for streamlining the text! @Pointedstick

**Comment by ravenshining** [comment]
Well, if Becca doesn't want to make sprites for the torpedo pods, I will:

https://github.com/Zitchas/endless-sky/pull/96
https://github.com/Zitchas/endless-sky-high-dpi/pull/4
https://drive.google.com/file/d/14cq9tH3DhkKPLAgO2gpn2BYZ1kp2phcb/view?usp=sharing

I'd disclaim the copyright but since Becca said he didn't want to make these I'll put them by me derived from becca in case saying becca did it would offend him.

**Comment by beccabunny** [comment]
It doesn't make sense though that more than half of a torpedo weights and takes up less than one third of the normal one, which is why I didn't make it

**Comment by ravenshining** [comment]
Sorry, I don't understand what you're trying to say.  It sounds to me like you have some concern about internal consistency, though I don't understand why that would stop you from making them.

Torpedo: 0.4 t
Torpedo Launcher: 13 t mass, 25 t space, 30 torpedoes
Torpedo rack: 3 t mass, 9t space, 15 torpedoes
30 * 0.4 t = 12 t worth of torpedoes in the launcher (checks out, 25 - 12 = 13)
15 * 0.4 t = 6 t worth of torpedoes in the rack (checks out, 9 - 6 = 3)
30 / 15 = 2 racks of torpedoes in a launcher
2 * 3 t = 6 tonnes worth of rack in a launcher
13 - 6 = 7 tonnes of launching mechanism

So to build a 5-torpedo pod, we need
7 tonnes or less of launching mechanism (pods have reduced ROF so at a guess, say... 5?)
9 / (15/5) = 3 tonnes of space and 3 / (15/5) = 1 tonne of mass for storage
5 + 3 = 8 tonnes, which happens to be exactly how big I guesstimated the pod would be in my PR to Zitchas
5 + 1 = 6 tonnes of mass, which is one tonnne of mass more than I guesstimated

So it all checks out to me.  If you're cool with it I'll gladly remove my own attribution in copyright and stick them under you.

**Comment by beccabunny** [comment]
I am talking about the image, not the data

**Comment by ravenshining** [comment]
You're going to have to elaborate, I have zero clue what you're talking about.  Or why it matters, it's not like the thumbnail shows the entire launcher with all 30 or 5 torpedoes in.

**Comment by beccabunny** [comment]
> 
> 
> You're going to have to elaborate, I have zero clue what you're talking about. Or why it matters, it's not like the thumbnail shows the entire launcher with all 30 or 5 torpedoes in.

This is your torpedo pod:
immagine
While this is the normal launcher:
immagine
Not only they are evidently the same thing cut up exactly in half, but they are also unfitting with the relative space they take up, as the launcher is 25 outfit space while the relative pod is down to 8 outfit space. Outfit sprites shoud be a good representation of their outfit, which in this case means the changes relative to the "normal" torpedo launcher, and erasing half of the launcher when the outfit is cut to less than 1/3 of its original space is not a good way to do it IMO. Regarding your second argument, please take a look at the other sprites, for example the sidewinder which changed this way:
immagine
immagine
I.e. just because the sprite may not show *all* of the ammunitions contained in the launcher that doesn't mean care shouldn't be put in correctly representing the change in size, as otherwise we could just put the sprites of the normal launchers and say "It was the rest of the launcher that was removed, so it looks the same to you". Visuals are important.

**Comment by ravenshining** [comment]
Sure, they're important, but that sounds like an awfully nitpicky reason to stop a perfectly good outfit idea.  Especially not for an outfit sprite which has no mechanical impact on the gameplay.  I think it's more important to have *something* that looks reasonable and communicates the general idea than to have it look proportionally perfect, especially when others are obviously missing elements, too.  If someone wants to improve the sprite down the line, they always can.

**Comment by beccabunny** [comment]
> Sure, they're important, but that sounds like an awfully nitpicky reason to stop a perfectly good outfit idea.

An image not being good for an outfit isn't a nitpicking, just because you may not care about their appearance that doesn't change the fact that they are the first thing you see and one of the most important parts of an outfit. Hearing that since the image doesn't have an impactful gameplay impact it doesn't matter and criticising it is nitpicking is honestly insulting for someone who did most of his work on outfit sprites. This one is just a bad representation for it, which is again why I didn't want to do it, as it isn't just a matter of spending a minute in blender deleting half of the objects and pressing render

**Comment by ravenshining** [comment]
It's a perfectly good image to me, just as good as any of yours.  Far more detailed than it has to be IMO, but that's besides the point.  Whether you choose to be insulted by my priorities is not relevant to this discussion.  If you have any better ideas of how to make a smaller sprite, that would be a productive direction to take this conversation.

**Comment by ravenshining** [comment]
Like, here, this would be one way of making it shorter.

Screenshot_20210416_115637

**Comment by Amazinite** [comment]
Taking the rocket launcher and pod as examples, increasing the apparent size of the torpedo launcher also seems to be an option instead of just shrinking the pod. Although I wouldn't say that the solution is as simple as the torpedo pod being one tube while the torpedo launcher is three, because otherwise it's just the same deal as the rocket launcher and pod. So making the pod appear even smaller is probably a better approach, and as far as that goes the shorter barrel sounds like a good idea.

Something else we might do is model the torpedo pod after an underbarrel grenade launcher. The M203 [for example](https://youtu.be/DFz36hpmVIQ?t=269) has the barrel pull forward during loading, then you pull it back onto the grenade. Applying that example here, the back end of the torpedo pod would be more open than it is right now, instead of just being open on the top where I assume that is where torpedoes are currently loaded from. So I imagine the barrel would need to be at least as long as the torpedo so that when the action is closed you don't have the torpedo sticking out the barrel, but the reduction in material around the back of the outfit where the torpedo currently is would certainly make it appear lighter.

**Comment by ravenshining** [comment]
Those are some great ideas, Derpy.  I hope a skilled blenderer can take those ideas and make an improved sprite.

I, however, am *not* a skllled blenderer, at least not enough to create an outfit from scratch that meets the new standard.  I can edit a model to achieve a satisfactory result - it is a torpedo launcher, it meets the visual standard set by the rest, it is smaller than the other one - but for me to redesign the thing would be a days- or weeks- long undertaking where a more skilled blenderer could probably do it in under an hour.

I am not opposed to the idea of outfits being perfectly scaled to one another.  I can appreciate the value in doing so, as an added bonus or flair.  However, I think this is an undue burden to place on all contributors, and should not be a matter of *policy* that prevents an outfit from becoming reality in the first place.

**Comment by Zitchas** [comment]
I merged the torpedo PRs into this as I think they are good enough to warrant inclusion.

Having a shorter torpedo pod (or alternately, a larger launcher) are both good ideas, and I hope someone makes one. If I can figure out blender again, I'll render the shorter tube that Ravenshining did and add that in. But that falls into the realm of "making things better" and not "absolutely required."

**Comment by TheMarksman-ES** [comment]
No observed errors or typos in the codes.

File: data/human/weapons.txt [hunk 2]
```diff
@@ -693,6 +693,45 @@ outfit "Meteor Missile Launcher"
...
+		icon "icon/meteor"
+		"fire effect" "meteor fire"
+		"die effect" "missile death"
+		"hit effect" "missile hit"
+		"inaccuracy" 5
+		"velocity" 11
+		"lifetime" 350
+		"reload" 200
+		"burst count" 7
+		"burst reload" 60
```
Comment (Pointedstick) [inline data/human/weapons.txt:720]: All the other pods have a `burst reload` value that's equal to the `reload` value of the base launcher, but this one does not. Recommend following the pattern:

```suggestion
		"burst reload" 80
```

File: data/human/weapons.txt [hunk 3]
```diff
@@ -866,6 +978,45 @@ outfit "Torpedo Launcher"
...
 
+outfit "Torpedo Pod"
+	category "Secondary Weapons"
+	cost 30000
+	thumbnail "outfit/torpedopod"
+	"mass" 5
+	"outfit space" -8
+	"weapon capacity" -8
+	"gun ports" -1
+	"torpedo capacity" 5
```
[the remark this answers is not in the document]
Comment (Zitchas) [reply data/human/weapons.txt:989]: Yes, definitely. Not sure what I was thinking with having it at 5. Changed.

File: data/human/weapons.txt [hunk 4]
```diff
@@ -928,6 +1079,45 @@ outfit "Typhoon Launcher"
...
 
+outfit "Typhoon Pod"
+	category "Secondary Weapons"
+	cost 100000
+	thumbnail "outfit/typhoonpod"
+	"mass" 6
+	"outfit space" -12
+	"weapon capacity" -12
+	"gun ports" -1
+	"typhoon capacity" 5
```
[the remark this answers is not in the document]
Comment (Zitchas) [reply data/human/weapons.txt:1090]: Likewise. Changed to 2.

**Comment by Zitchas** [comment]
@Pointedstick Thanks for catching those erroneous capacities on the torpedoes. Those have been fixed now.

**Review by Pointedstick — APPROVED** [review]
LGTM now

**Comment by WarlordMike** [comment]
I remember seeing this the first time it was mentioned on the Discord, and I would love to see this implemented. Might even lead to some rebalancing of outfit spaces on drones/fighters/interceptors.

**Review by WarlordMike — APPROVED** [review]

**Comment by Pointedstick** [comment]
I have a slight concern that the meteor, sidewinder, and javelin pods don't seem to offer a significant advantage over the full-sized launchers with the way fighters and interceptors work today. They can generally fit one or even two of the full-sized versions already (or 3-4 in the case of the Fury!). For these ships, using a pod instead of a launcher simply reduces missile capability in exchange for less space consumed. But what are you going to fill that now-empty space with?

A single small gun is not useful; it would make more sense to just use all-missile armament with full-sized pods instead. So you would probably fill that extra space with more, different missile pods. Like a sidewinder pod and a heavy rocket pod. Actually now that I think about it, that *is* a useful thing.

I think this change will probably make additional sense in conjunction with other changes that either reduce fighter/interceptor weapon space or give them more gun ports--or both. And other PRs to adjust the default loadouts to take advantage of these new pods.

OK, let's do it. We've tortured this concept to death long enough. :) We can adjust loadouts and small ship capacities (if needed) in subsequent PRs.

**Comment by Zitchas** [comment]
In some ways, these pods are a continuation of the idea that nuclear missiles introduced: If warships can carry big missiles that are so big that a single projectile takes up the entire hardpoint, why can't smaller ships similarly equip missiles that are just as big to them? A torpedo on a fighter probably feels just as space consuming as a nuclear missile on a light warship. 

Pods also tie in really well with the recent "reload from mothership" mechanic. Up until now, there hasn't really been a use for it, as reloading fighters with large inventories generally felt redundant to me. Also risky - fighters die really fast. Do I *really* want to give them 30 missiles so they can take a bunch of my inventory with them when they go off to die? on the other hand, being able to dole out a handful to each fighter in each wave feels like a much smarter way to do it.

These pods open up a whole range of possible changes, and while I'm sure I'll PR some of them, I hope others do to. 
Some possibilities that I foresee:
- Changes that involve having merchant ships be less "warships with cargo" and more "merchants that can defend themselves"
- Fighters become more fighters and less "mini-warships"
- Possibility of introducing bombers as a separate kind of fighter
- Smaller, lighter, fighters (and thus faster too)

**Comment by Pointedstick** [comment]
Yeah those changes make sense. Honestly my favorite proposed change was to make carried fighters into purchaseable and non-customizable outfits. The idea would be that the ships needed to be miniaturized by hardcoding their systems to make them small enough to carry them onboard. So interceptors would be the non-carryable yet customizable and hyperspace-capable versions. I thought that idea really had some merit. With this, outfitting such vessels with pods would make perfect sense given the lore reasoning of the need to save space and weight.

I also liked the idea of instead making *all* missiles carryable on hardpoints, not only nuclear missiles. With that change, fighters and interceptors would simply be given 10-15 gun ports, with no increase in total weapon capacity. So you could outfit a fighter-comber by outfitting a mix of sidewinders, torpedoes, and heavy rockets, with one or two guns as backup. That's something you still can't do today even with these new missile pods, due to the very low number of gun hardpoints on most fighter and interceptor sized craft. I think that needs to be one of the first things that changes. That was proposed in https://github.com/endless-sky/endless-sky/pull/3350 and I was sad that it didn't end up happening.

**Comment by Hecter94** [comment]
I have a question regarding this PR. Is there a specific reason why the Javelin Mini Pod and the Heavy Rocket Pod have a range override set different from their actual range? 
Heavy Rocket Pod has a range of 1000 but has a range override of 500; Javelin Mini-Pod has a range of 600 but a range override of 300. 

This means they display in the outfitter as having different ranges from their full-size launcher counterpart, despite them having the same range:
<p float="left">
  
   
</p>

image

Is there a specific reason why range override needed to be used here? It seems less than ideal to be displaying erroneous information to the player.

**Comment by Zitchas** [comment]
The main reason those have a range override is so that the AI doesn't waste ammo shooting them at max range when they have virtually zero chance to hitting anything. There's no reason the projectile itself can't go that far, so that's not a problem. And the player can definitely *try* to shoot things that far if they want. Just need to prevent the AI from trying for max range shots.

I don't know why they have different max ranges off the top of my head. My suspicion is that either
a) it's a mistake, and they should both have the same range override.
b) that it was decided to allow the larger capacity pods (typically on bigger ships) to utilize the full range since they have more ammo and the very high miss-rate at those ranges isn't as important.

**Comment by Hecter94** [comment]
Looks like this was resolved with #8664

**Pull request merged**